src/opers.c

```diff
@@ -642,53 +657,41 @@ Obj FuncAND_FLAGS (
             "you can replace <flags2> via 'return <flags2>;'" );
     }
 
+    if (flags1 == flags2)
+        return flags1;
+
     // check the cache
 #   ifdef AND_FLAGS_HASH_SIZE
         // We want to ensure if we calculate 'flags1 and flags2', then
         // later do 'flags2 and flags1', we will get the value from the cache.
         // Therefore we just compare the location of the Bag masterpointers
         // for both flags (which doesn't change), and use the cache of the
         // smaller.
-        if ( flags1 < flags2 ) {
-            flagsX = flags2;
-#           ifdef HPCGAP
-                if (!PreThreadCreation) {
-                    locked = flags1;
-                    HashLock(locked);
-                }
-#           endif
-            cache  = AND_CACHE_FLAGS(flags1);
-            if ( cache == 0 ) {
-                cache = NEW_PLIST( T_PLIST, 2*AND_FLAGS_HASH_SIZE );
-                MakeBagPublic(cache);
-                SET_AND_CACHE_FLAGS( flags1, cache );
-                CHANGED_BAG(flags1);
-            }
+        if ( flags1 > flags2 ) {
+            SWAP(Obj, flags1, flags2);
         }
-        else {
-            flagsX = flags1;
-#           ifdef HPCGAP
-                if (!PreThreadCreation) {
-                    locked = flags2;
-                    HashLock(locked);
-                }
-#           endif
-            cache  = AND_CACHE_FLAGS(flags2);
-            if ( cache == 0 ) {
-                cache = NEW_PLIST( T_PLIST, 2*AND_FLAGS_HASH_SIZE );
-                MakeBagPublic(cache);
-                SET_AND_CACHE_FLAGS( flags2, cache );
-                CHANGED_BAG(flags2);
+
+#       ifdef HPCGAP
+            if (!PreThreadCreation) {
+                locked = flags2;
+                HashLock(locked);
             }
+#       endif
+        cache  = AND_CACHE_FLAGS(flags2);
```

Comment:
Your comment above, and code here, don't seem to agree -- you say you get the cache of the smaller, but you get the cache of the bigger (I don't mine which gets fixed, as I don't think it matters).

Reply:
Oops, indeed, will fix.

Reply:
Fixed